src/main/java/com/yelp/nrtsearch/server/handler/SearchHandler.java

```diff
@@ -387,6 +388,12 @@ public SearchResponse handle(IndexState indexState, SearchRequest searchRequest)
   private void fetchFields(SearchContext searchContext)
       throws IOException, ExecutionException, InterruptedException {
     if (searchContext.getResponseBuilder().getHitsBuilderList().isEmpty()) {
+      if (searchContext.getFetchTasks().getHitsLoggerFetchTask() != null) {
+        searchContext
```

Comment:
can you leave a comment here that we may want to log request even if there is no hits t o return